packages/uma-auth-client/src/hooks/useOAuth.tsx

```diff
@@ -0,0 +1,100 @@
+import {auth, nwc, webln} from '@getalby/sdk';
+import { useEffect, useState } from "react";
+import { getUmaDomain } from "src/utils/getUmaDomain";
+import { useUser } from "./useUser";
+import { create } from 'zustand';
+import { persist } from "zustand/middleware";
+import { getNwcUmaDomain } from 'src/utils/getNwcUmaDomain';
+
+type UmaAuthToken = {
+  access_token: string,
+  refresh_token: string,
+  expires_at: number,
+  nwc_connection_uri: string;
+  commands: string[];
+  budget: string;
+  nwc_expires_at: string;
+};
+
+interface TokenState {
+  accessToken: string;
+  refreshToken: string;
+  expiresAt: number;
+}
+
+interface OAuthState {
+  codeVerifier?: string;
+  token?: TokenState;
+  identityNpub?: string;
+  identityRelayUrl?: string;
+  redirectUri?: string;
+  nwcConnectionUri?: string;
+  setToken: (token: TokenState) => void;
+  performOAuth: (uma: string) => Promise<void>;
+}
+
+export const useOAuth = create<OAuthState>()(
+  persist(
+    (set, get) => ({
+      setToken: (token) => set({
+        token,
+      }),
+      performOAuth: async (uma) => {
+        const state = get();
+        const result = await performOAuth(state, uma);
+        set(result);
+      },
+    }),
+    {
+      name: "uma-connect",
+      partialize: (state) => ({
+        nwcConnectionUri: state.nwcConnectionUri,
+      })
+    }
+  )
+);
+
+const performOAuth = async (state: OAuthState, uma: string) => {
+  const { identityNpub, identityRelayUrl, redirectUri, codeVerifier, token, nwcConnectionUri } = state;
```

Comment:
Yeah some of this stuff definitely needs to be passed in as a config, but I know we talked about doing that as a follow-up, which is fine.